api/management-model/src/test/java/org/apache/polaris/core/admin/model/CatalogSerializationTest.java

```diff
@@ -0,0 +1,172 @@
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        TEST_CATALOG_NAME,
+                        new CatalogProperties(TEST_LOCATION),
+                        new AwsStorageConfigInfo(
+                            TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3))),
+            new TestCase("Null fields")
+                .withCatalog(new Catalog(Catalog.TypeEnum.INTERNAL, null, null, null)),
+            new TestCase("Long name")
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        "a".repeat(1000),
+                        new CatalogProperties(TEST_LOCATION),
+                        null)),
+            new TestCase("Unicode characters")
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        "测试目录",
+                        new CatalogProperties(TEST_LOCATION),
+                        null)),
+            new TestCase("Empty strings")
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        "",
+                        new CatalogProperties(""),
+                        new AwsStorageConfigInfo("", StorageConfigInfo.StorageTypeEnum.S3))),
+            new TestCase("Special characters")
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        "test\"catalog",
+                        new CatalogProperties(TEST_LOCATION),
+                        new AwsStorageConfigInfo(
+                            TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3))),
+            new TestCase("Whitespace")
+                .withCatalog(
+                    new Catalog(
+                        Catalog.TypeEnum.INTERNAL,
+                        "  test  catalog  ",
+                        new CatalogProperties("  " + TEST_LOCATION + "  "),
+                        null)));
+
+    Stream<TestCase> arnCases =
+        Stream.of(
+                "arn:aws:iam::123456789012:role/test-role",
+                "arn:aws:iam::123456789012:role/service-role/test-role",
+                "arn:aws:iam::123456789012:role/path/to/role")
+            .map(
+                arn ->
+                    new TestCase("ARN: " + arn)
+                        .withCatalog(
+                            new Catalog(
+                                Catalog.TypeEnum.INTERNAL,
+                                TEST_CATALOG_NAME,
+                                new CatalogProperties(TEST_LOCATION),
+                                new AwsStorageConfigInfo(
+                                    arn, StorageConfigInfo.StorageTypeEnum.S3))));
+
+    return Stream.concat(basicCases, arnCases);
+  }
+
+  private static class TestCase {
```

Comment:
Why this class and not just parameterized-test arguments?

Reply:
I agree with @snazy .

Please consider producing individual `Catalog` objects as test method arguments (and remove `TestCase`).

If you prefer to have hand-written test case descriptions, please consider using `ArgumentSet` (from JUnit5).